gsl_programmation/services.py

```diff
@@ -1,14 +1,55 @@
 from decimal import Decimal
+from typing import Any
 
 from django.db.models import Case, F, Q, Sum, When
 from django.db.models.query import QuerySet
 
+from gsl_core.models import Perimetre
 from gsl_demarches_simplifiees.models import NaturePorteurProjet
-from gsl_programmation.models import Simulation, SimulationProjet
+from gsl_programmation.models import Enveloppe, Simulation, SimulationProjet
+from gsl_programmation.utils import slugify
```

Comment:
Tu peux utiliser l'utilitaire Django directement : 

```suggestion
from django.utils.text import slugify
```

Reply:
rah, mon cmd + maj + ; ne me le proposait pas !